sketch-hq/sketch-assistants
# Add prefer color variables rule

- Adds a new rule to the Core Assistant: `colors-prefer-variables`
- Makes use of the new rule in the Reuse Suggestions Assistants

### Next steps

There will still be steps to complete once this PR is merged:

1. Merge CrowdIn translations for the 4 new strings added as part of this PR
1. After that, release the new Assistants based on the changesets included in this PR

### Screenshot

<img width="75%" src="https://user-images.githubusercontent.com/1078571/93495839-7964f180-f906-11ea-988e-d7c99b2b0be6.jpg" />

### Related issues

- Refs #148 

### QA Instructions

1. Check out this branch
1. From the root run `yarn build`
1. Change to the Reuse Suggestions Assistant package `cd assistants/reuse-suggestions`
1. Build its tarball `yarn package-tarball`
1. Add the tarball to a Sketch 69+ document

The Reuse Suggestion Assistant should complain about more than 2 usages of a hard coded color, suggesting to use a color variable instead. It checks colors in the following places:

- Artboard, Slice and Symbol Master backgrounds
- Borders, Fills, Shadows and Inner Shadows of layer styles
- Colors applied to text strings
